test/dynamo/test_misc.py

```diff
@@ -4443,6 +4443,58 @@ def getitem(a, idx):
         self.assertTrue(ref1 == res1)
         self.assertTrue(ref2 == res2)
 
+    def test_trace_einops(self):
```

Comment:
Discussed plan:
- we add an config.enable_einops_tracing
- we are going to add an extra list somewhere "dont_allow_in_graph"
- if config.enable_einops_tracing then we put einops functions into "dont_allow_in_graph" 
- trace_rules.py checks this list when it is creating a TorchInGraphVariable. If something is in the list, and about to become a TorchInGraphVariable, we make it something else. This solves the "import order issue"
- new torch old einops case: in this case, if config.enable_einops_tracing is false then we rely on either: (1) einops doing allow_in_graph (if the version is recent enough), or (2) we will manually do the allow in graph (if the version is old enough). There already exists a switch somewhere for this in Dynamo

Reply:
branch cut is 6/23